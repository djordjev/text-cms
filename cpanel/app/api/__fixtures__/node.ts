import { FsNode } from '@prisma/client';

const newNode = (prototype?: Partial<FsNode>, children?: Partial<FsNode>[]) => {
  // eslint-disable-next-line
  const result: any = {
    createdAt: prototype?.createdAt ?? new Date(2024, 2, 6, 12, 12, 12, 12),
    id: prototype?.id ?? 1,
    name: prototype?.name ?? 'random_name',
    parentId: prototype?.parentId ?? 1001,
    path: prototype?.path ?? '/randompath',
    typeId: prototype?.typeId ?? 1
  };

  if (children) {
    result.children = [];

    for (const child of children) {
      const newChild = {
        createdAt: child?.createdAt ?? new Date(2024, 2, 6, 12, 12, 12, 12),
        id: child?.id ?? 1,
        name: child?.name ?? 'random_name',
        parentId: child?.parentId ?? 1001,
        path: child?.path ?? '/randompath',
        typeId: child?.typeId ?? 1
      };

      result.children.push(newChild);
    }
  }

  return result;
};

export { newNode };
