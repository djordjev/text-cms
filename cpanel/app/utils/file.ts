const isHome = (name: string) => {
  return name.toLowerCase() === 'finder';
};

const isFile = (name: string) => {
  return name.endsWith('.txt');
};

const isFolder = (name: string) => {
  return !isFile(name);
};

const getPathFromSegments = (segments: string[], segment: string) => {
  const index = segments.indexOf(segment);

  if (index < 1) return '/finder';

  const mapped = segments.slice(0, index + 1).map((segment) => {
    if (isHome(segment)) return 'finder';

    return segment;
  });

  const joined = mapped.join('/');

  return `/${joined}`;
};

export { getPathFromSegments, isHome, isFile, isFolder };
