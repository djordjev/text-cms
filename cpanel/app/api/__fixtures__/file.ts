import { TEST_EDITOR } from '~/utils/test';

export const FILE_CONTENT = [
  {
    id: '1',
    name: 'First Variation',
    condition: [[['a', '=', '1']]],
    text: TEST_EDITOR
  },
  {
    id: '2',
    name: 'Second Variation',
    condition: [
      [
        ['b', '=', '1'],
        ['c', '>', '12']
      ],
      [['w', '=', 'true']]
    ],
    text: TEST_EDITOR
  }
];

export const FILE_VARIATION = JSON.stringify(FILE_CONTENT);
