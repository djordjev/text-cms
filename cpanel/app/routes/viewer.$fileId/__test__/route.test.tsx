vi.mock('~/utils/env');

import { createRemixStub } from '@remix-run/testing';
import { render, screen } from '@testing-library/react';

import { FILE_CONTENT } from '~/api/__fixtures__/file';
import { newNode } from '~/api/__fixtures__/node';

import Viewer from '../route';

test('viewer route', async () => {
  const createComponent = () => {
    const RemixStub = createRemixStub([
      {
        path: '/viewer/:fileId',
        Component: Viewer,
        loader: () => {
          return {
            info: newNode({ id: 22, name: 'File 1' }),
            variations: FILE_CONTENT
          };
        }
      }
    ]);

    return <RemixStub initialEntries={['/viewer/22']} />;
  };

  const { container } = render(createComponent());

  expect(
    await screen.findByRole('link', { name: 'Add new variation' })
  ).toBeInTheDocument();

  FILE_CONTENT.forEach((fc) =>
    expect(screen.getByText(fc.name)).toBeInTheDocument()
  );

  expect(container.childNodes).toMatchSnapshot();
});
