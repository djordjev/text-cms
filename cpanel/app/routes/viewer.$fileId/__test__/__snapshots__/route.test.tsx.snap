// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`viewer route 1`] = `
NodeList [
  <div
    class="u-w-full"
  >
    <div
      class="u-text-lg u-w-full u-flex u-justify-between u-items-center"
    >
      <div
        class="u-mr-2xs"
      >
        <span
          class="u-text-primary"
        >
          File:
        </span>
        <span
          class="u-text-primary-100 u-font-bold"
        >
          /randompath
        </span>
      </div>
      <div>
        <button
          class="u-btn u-btn-outline u-mr-1x"
          type="button"
        >
          Reorder
        </button>
        <a
          class="u-btn u-btn-accent"
          href="/editor/22"
        >
          Add new variation
        </a>
      </div>
    </div>
    <div
      class="u-p-4x"
    >
      <div
        class="u-card u-shadow-xl u-mb-5x u-bg-base-300 u-p-3xs"
      >
        <div
          class="u-mb-1x u-flex"
        >
          <div
            class="u-flex-1"
          >
            <span
              class="u-italic"
            >
              Variation Name: 
            </span>
            <h4
              class="u-text-secondary-content u-font-bold u-inline"
            >
              First Variation
            </h4>
          </div>
          <div
            class="u-flex u-gap-1x"
          >
            <form
              action="/viewer/22"
              method="post"
            >
              <button
                class="u-link u-link-primary"
                type="button"
              >
                <svg
                  class="u-inline"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 7l16 0"
                  />
                  <path
                    d="M10 11l0 6"
                  />
                  <path
                    d="M14 11l0 6"
                  />
                  <path
                    d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
                  />
                  <path
                    d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
                  />
                </svg>
                 Delete
              </button>
            </form>
            <a
              class="u-link u-link-primary"
              href="/editor/22/1"
            >
              <svg
                class="u-inline"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 7h-1a2 2 0 0 0 -2 2v9a2 2 0 0 0 2 2h9a2 2 0 0 0 2 -2v-1"
                />
                <path
                  d="M20.385 6.585a2.1 2.1 0 0 0 -2.97 -2.97l-8.415 8.385v3h3l8.385 -8.415z"
                />
                <path
                  d="M16 5l3 3"
                />
              </svg>
               Edit
            </a>
          </div>
        </div>
        <div
          class="u-flex u-mb-1x"
        >
          <span
            class="u-mr-3x u-italic"
          >
            Conditions:
          </span>
          <div>
            <div>
              <span>
                a
                 
                =
                 
                1
              </span>
              <span
                class="u-font-bold u-text-secondary-content last:u-hidden"
              >
                 
                AND
                 
              </span>
            </div>
            <div
              class="u-divider last:u-hidden u-my-1xs"
            >
              OR
            </div>
          </div>
        </div>
        <div>
          <h5
            class="u-italic u-mb-2xs"
          >
            Text:
          </h5>
          <div>
            <h3
              class="u-text-center u-text-lg"
            >
              <span
                class=""
              >
                HEADING
              </span>
            </h3>
            <p
              class=""
            >
              <span
                class=""
              >
                A 
              </span>
              <span
                class="u-italic"
              >
                line
              </span>
              <span
                class=""
              >
                 of 
              </span>
              <span
                class="u-font-bold"
                style="color: rgb(228, 17, 17);"
              >
                text
              </span>
              <span
                class=""
              >
                 in a 
              </span>
              <span
                class="u-font-bold"
              >
                paragraph
              </span>
              <span
                class=""
              >
                .
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="u-card u-shadow-xl u-mb-5x u-bg-base-300 u-p-3xs"
      >
        <div
          class="u-mb-1x u-flex"
        >
          <div
            class="u-flex-1"
          >
            <span
              class="u-italic"
            >
              Variation Name: 
            </span>
            <h4
              class="u-text-secondary-content u-font-bold u-inline"
            >
              Second Variation
            </h4>
          </div>
          <div
            class="u-flex u-gap-1x"
          >
            <form
              action="/viewer/22"
              method="post"
            >
              <button
                class="u-link u-link-primary"
                type="button"
              >
                <svg
                  class="u-inline"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 7l16 0"
                  />
                  <path
                    d="M10 11l0 6"
                  />
                  <path
                    d="M14 11l0 6"
                  />
                  <path
                    d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
                  />
                  <path
                    d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
                  />
                </svg>
                 Delete
              </button>
            </form>
            <a
              class="u-link u-link-primary"
              href="/editor/22/2"
            >
              <svg
                class="u-inline"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 7h-1a2 2 0 0 0 -2 2v9a2 2 0 0 0 2 2h9a2 2 0 0 0 2 -2v-1"
                />
                <path
                  d="M20.385 6.585a2.1 2.1 0 0 0 -2.97 -2.97l-8.415 8.385v3h3l8.385 -8.415z"
                />
                <path
                  d="M16 5l3 3"
                />
              </svg>
               Edit
            </a>
          </div>
        </div>
        <div
          class="u-flex u-mb-1x"
        >
          <span
            class="u-mr-3x u-italic"
          >
            Conditions:
          </span>
          <div>
            <div>
              <span>
                b
                 
                =
                 
                1
              </span>
              <span
                class="u-font-bold u-text-secondary-content last:u-hidden"
              >
                 
                AND
                 
              </span>
              <span>
                c
                 
                &gt;
                 
                12
              </span>
              <span
                class="u-font-bold u-text-secondary-content last:u-hidden"
              >
                 
                AND
                 
              </span>
            </div>
            <div
              class="u-divider last:u-hidden u-my-1xs"
            >
              OR
            </div>
            <div>
              <span>
                w
                 
                =
                 
                true
              </span>
              <span
                class="u-font-bold u-text-secondary-content last:u-hidden"
              >
                 
                AND
                 
              </span>
            </div>
            <div
              class="u-divider last:u-hidden u-my-1xs"
            >
              OR
            </div>
          </div>
        </div>
        <div>
          <h5
            class="u-italic u-mb-2xs"
          >
            Text:
          </h5>
          <div>
            <h3
              class="u-text-center u-text-lg"
            >
              <span
                class=""
              >
                HEADING
              </span>
            </h3>
            <p
              class=""
            >
              <span
                class=""
              >
                A 
              </span>
              <span
                class="u-italic"
              >
                line
              </span>
              <span
                class=""
              >
                 of 
              </span>
              <span
                class="u-font-bold"
                style="color: rgb(228, 17, 17);"
              >
                text
              </span>
              <span
                class=""
              >
                 in a 
              </span>
              <span
                class="u-font-bold"
              >
                paragraph
              </span>
              <span
                class=""
              >
                .
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
